### PR TITLE
added libvirt to the vagrantfile template

### PR DESCRIPTION
**Pull request by oserde**
Hi. I am using libvirt as provider, and added this lines for being able to set and configure libvirt from the molecule.yml file. It is not a big deal, but including these lines I would save copying a specific template for libvirt to the molecule templates directory each time I install molecule(which is quite frequent) on my virtual environments.

Thank you.

### CODE REVIEW

**Comment by dhutty** [comment]
Please add to docs/source/providers.rst to document this support for libvirt as an alternate provider.

**Comment by abrown-sg** [comment]
@oserde Yeah, would definitely like to see an example of what the corresponding provider looks like in molecule.yml.

It's nice to see it was possible to get another provider working relatively easily!

**Comment by oserde** [comment]
I plan to add some features to improve libvirt support. So in a few days I will add the documentation and these new features as well.

**Comment by retr0h** [comment]
Awaiting @oserde to update PR before we merge.

**Comment by oserde** [comment]
Ok guys. I will have it in a few hours.

**Comment by oserde** [comment]
mmm...still need to fix the doc urls

**Comment by oserde** [comment]
done

File: docs/source/providers.rst [hunk 1]
```diff
@@ -179,3 +179,112 @@ Other Notes
...
+	      options:
+		append_platform_to_hostname: no
+
+	      ansible_groups:
+		- group_1
+
+Notes
+^^^^^
+
+* Molecule destroy does not seem to work well. It removes some molecule files but not always destroys de VM. You may have to do it manually by using the virsh command.
```
Comment (dhutty) [inline docs/source/providers.rst]: Please raise an [Issue](https://github.com/metacloud/molecule/issue) with details about what you experience when molecule destroy does not work for you.

**Comment by oserde** [comment]
#123

**Comment by oserde** [comment]
By the way, I use:
- Ubuntu 14.04
- libvirtd (libvirt) 1.2.2
- Vagrant 1.7.0 with the vagrant-libvirt (0.0.32) plugin
- molecule 1.3.0

Despite the issue I previously referenced, vagrant-libvirt seems to work quite nicely on my machine.

**Comment by abrown-sg** [comment]
@oserde We haven't forgotten you here. We're working on an update to Vagrantfile that will change its behavior somewhat. I'll try to incorporate your changes here as part of that.

**Comment by oserde** [comment]
ok, thanks. By the way, are you going to incorporate openstack as well?

**Comment by conorsch** [comment]
Status update? I've been using libvirt a lot lately, and since molecule doesn't officially support it, I'm using a custom Vagrantfile template to add the support.

**Comment by retr0h** [comment]
I don't know when we will get around to performing the refactor.  I'm happy to accept a new PR for this, and we'll just add it in.

**Comment by conorsch** [comment]
Sure, I'd be happy to repackage these changes as a new PR. @oserde already did all the work, so I'll make sure to credit appropriately in the commits.

**Comment by retr0h** [comment]
In the future I'd like to rework the Vagrantfile to something like [this](https://github.com/retr0h/maquina/blob/master/maquina/driver.py#L157-L175).  We can then plugin in provisioners a lot easier.

**Comment by conorsch** [comment]
Hey, thanks for pointing out maquina, that definitely looks more readable and maintainable than the jinja nightmare the vagrantfile template has already become. Will take a closer look at maquina!

**Comment by retr0h** [comment]
Closing in favor of #200.

**Pull request closed without merging**